components/ActivityCard.js

```diff
@@ -1,9 +1,16 @@
 import Image from "next/image";
 import styled from "styled-components";
+import { useRouter } from "next/router";
+
+export default function ActivityCard({ id, title, categories, imageUrl }) {
+  const router = useRouter();
+
+  const handleClick = () => {
+    router.push(`/activity/${id}`);
+  };
 
-export default function ActivityCard({ title, categories, imageUrl }) {
   return (
-    <StyledCardSection>
+    <StyledCardSection onClick={handleClick}>
```

Comment:
If the entire card is intended as a link, bookmarking a specific activity could become challenging, especially as this button will likely be on the card later and may trigger a redirect when clicked.
Possible solutions: Use the activity's name or the activity image as a link.